textproto/mod.ts

```diff
@@ -126,39 +122,29 @@ export class TextProtoReader {
   }
 
   async readLineSlice(): Promise<Uint8Array | null> {
-    // this.closeDot();
-    let line: Uint8Array | undefined;
-    while (true) {
-      const r = await this.r.readLine();
-      if (r === null) return null;
-      const { line: l, more } = r;
-
-      // Avoid the copy if the first call produced a full line.
-      if (!line && !more) {
-        // TODO(ry):
-        // This skipSpace() is definitely misplaced, but I don't know where it
-        // comes from nor how to fix it.
-        if (this.skipSpace(l) === 0) {
-          return new Uint8Array(0);
-        }
-        return l;
-      }
-      line = line ? concat(line, l) : l;
-      if (!more) {
-        break;
+    let line: Uint8Array = new Uint8Array(0);
+    let r: ReadLineResult | null = null;
+
+    do {
+      r = await this.r.readLine();
+      if (r !== null && this.skipSpace(r.line) !== 0) { //TODO: Kept skipSpace to preserve behavior but it should be looked into to check if it makes sense when this is used.
+        line = concat(line, r.line);
       }
-    }
-    return line;
+    } while (r !== null && r.more);
+
+    return r === null ? null : line;
   }
 
   skipSpace(l: Uint8Array): number {
+    const charCodes: Array<number> = [CHAR_SPACE, CHAR_TAB];
     let n = 0;
-    for (let i = 0; i < l.length; i++) {
-      if (l[i] === charCode(" ") || l[i] === charCode("\t")) {
-        continue;
+
+    for (const val of l) {
+      if (!charCodes.includes(val)) {
+        ++n;
```

Comment:
Is there a reason for not using `n++` here?

Reply:
Personal discipline and preference really. Using pre-increment tends to lead to a safer code base than post increment. In this case it makes no difference. If the norm of this repository is to use post increment I have no problem changing it.

Reply:
The post-increment form is widely used, and most people would consider it the "default"; using the pre-increment form here makes me ask myself "why?" "is there something I'm not seeing here?". I also have my doubts about it being safer, but that's secondary to the first point.

Reply:
I changed it to `n++` as it seems to bother you (and thus may bother other people).
A PR comment section does not seem to be the proper place to have a debate about post-increment and pre-increment but I would be interested in exchanging opinions on the subject. You can DM me on Reddit or e-mail me about it. 👍